astroid/brain/brain_typing.py

```diff
@@ -344,6 +346,34 @@ def infer_tuple_alias(
     return iter([class_def])
 
 
+def _looks_like_typing_cast(node: Call) -> bool:
+    return isinstance(node, Call) and (
+        isinstance(node.func, Name)
+        and node.func.name == "cast"
+        or isinstance(node.func, Attribute)
+        and node.func.attrname == "cast"
+    )
+
+
+def infer_typing_cast(
+    node: Call, ctx: context.InferenceContext = None
+) -> typing.Iterator[Instance]:
+    """Infer call to cast() returning an instance of the cast-to type"""
+    try:
+        func = next(node.func.infer(context=ctx))
+    except InferenceError as exc:
+        raise UseInferenceDefault from exc
+    if func.qname() != "typing.cast" or len(node.args) != 2:
+        raise UseInferenceDefault
+
+    peek_value_node, value_node = itertools.tee(node.args[1].infer(context=ctx))
+    if next(peek_value_node) is Uninferable:
+        type_node = next(node.args[0].infer(context=ctx))
+        return iter([Instance(type_node)])
+    else:
+        return value_node
```

Comment:
`else` is redundant and can be removed since the `if` clause ends with `return`. That's the `pylint` error.

```py
    return iter([inferred])
```